build.xml

```diff
@@ -9,39 +9,50 @@
   <property name="skip.tests" value="true"/>
    -->
 
-	<property name="version" value="0.0.0" />
+	<exec executable="git" outputproperty="git.revision">
+		<arg value="describe" />
+		<arg value="--always" />
+	</exec>
+	<property name="version" value="0.0.0-${git.revision}" />
 
-	<property name="build.dir" value="./bin/main" />
-	<property name="build.dir.compile" value="${build.dir}/" />
-	<property name="build.dir.resources" value="${build.dir.compile}" />
+	<property name="build.dir" value="./bin" />
+	<property name="build.dir.main" value="${build.dir}/main" />
+	<property name="build.dir.main.compile" value="${build.dir.main}/" />
+	<property name="build.dir.main.resources" value="${build.dir.main.compile}" />
 
 	<!-- Tests -->
 	<property name="test.groups" value="" />
 	<property name="test.excludegroups" value="long-runtime" />
-	<property name="test.build.dir" value="./bin/test"/>
-	<property name="test.build.dir.report" value="./test-report" />
-	<property name="test.build.dir.compile" value="${test.build.dir}" />
-  <property name="test.build.dir.resources" value="${test.build.dir.compile}" />
+	<property name="build.dir.test" value="./bin/test"/>
+	<property name="build.dir.test.report" value="./test-report" />
+	<property name="build.dir.test.compile" value="${build.dir.test}" />
+	<property name="build.dir.test.resources" value="${build.dir.test.compile}" />
 
-	<!-- src directories --> 
+	<!-- src directories -->
 	<property name="src.dir" value="./src" />
-  <property name="src.dir.java" value="${src.dir}/main/java" />
```

Comment:
Check out user preferences in sublime to ensure they stick (if you are in sublime).  If in eclipse, then ensure the ./build/cruise_preferences.xml is loaded.

    {
      "ensure_newline_at_eof_on_save": true,
      "font_size": 15,
      "ignored_packages":
      [
        "Vintage"
      ],
      "tab_size": 2,

      "theme": "Afterglow-green.sublime-theme",
      "color_scheme": "Packages/Theme - Afterglow/Afterglow.tmTheme",
      "tabs_small": true,
      "status_bar_brighter": true,

      // "theme": "Spacegray Eighties.sublime-theme",
      // "color_scheme": "Packages/Theme - Spacegray/base16-eighties.dark.tmTheme",

      "translate_tabs_to_spaces": true,
      "trim_trailing_white_space_on_save": true
    }